src/link/C/zig.h

```diff
@@ -502,8 +502,20 @@ zig_shl_sat_s(int,        int, ((sizeof(int     )) * CHAR_BIT - 1))
 zig_shl_sat_s(long,      long, ((sizeof(long    )) * CHAR_BIT - 1))
 
 #define zig_bitsizeof(T) (CHAR_BIT * sizeof(T))
+#define zig_bit_mask(T, bit_width) \
+    ((bit_width) == 128 \
+     ? ((T)(((uint128_t) 0xffffffffffffffff) << 64 | \
+             (uint128_t) 0xffffffffffffffff)) \
+     : (((T)1 << (T)(bit_width)) - 1))
+
+static inline uint128_t zig_sign_extend(uint128_t value, uint128_t zig_type_bit_width) {
+    const uint128_t m = (uint128_t)1 << (zig_type_bit_width - 1);
+    return (value ^ m) - m;
+}
+
+static inline int zig_clz(unsigned long long value, uint8_t zig_type_bit_width, bool signed_type) {
```

Comment:
I added this parameter here (and in `zig_ctz()` and `zig_popcount()` below) even though it's unused so that all these function signatures are identical to each other and can be invoked with the same code in `airBuiltinCall()`.

If this is not acceptable, I'll make passing the parameter optional in `airBuiltinCall()` so that it is selectively emitted. WDYT?

Reply:
I think we should make decisions based on the quality of the output C code. Looking at the output C code, one would think, "why is clz defined to take a parameter that it never uses?" and there is not really a good answer for that. `zig_clz` should not take a parameter that it does not use. The implementation of the compiler serves to produce the C code; not the other way around.

Reply:
That sounds fair. I added an option to `airBuiltinCall()` so that it can output the signedness information only when needed. Let me know if you think it looks good now.

Reply:
With the latest changes you requested, this issue solved itself -- marking the discussion thread resolved.